robot/server/server.go

```diff
@@ -67,10 +71,24 @@ func (s *Server) Tunnel(srv pb.RobotService_TunnelServer) error {
 		return fmt.Errorf("failed to receive first message from stream: %w", err)
 	}
 
+	var destAllowed bool
+	dialTimeout := defaultTunnelConnectionTimeout
+	// Ensure destination port is available; otherwise error.
+	for _, tte := range s.robot.TrafficTunnelEndpoints() {
+		if int(req.DestinationPort) == tte.Port {
+			destAllowed = true
+			if tte.ConnectionTimeout != 0 {
+				dialTimeout = tte.ConnectionTimeout
+			}
+		}
+	}
+	if !destAllowed {
+		return fmt.Errorf("tunnel not available at port %d", req.DestinationPort)
```

Comment:
Planning on manually testing that this error path is correctly reached and that timeouts are correctly applied. Automated testing will be done as part of RSDK-9852.

Reply:
Manually tested; error path is correctly reached and timeout is correctly applied. This code is now commented out; I'll comment in back in as part of RSDK-9763 after the deployed `app` version upgrades its `rdk` dependency.